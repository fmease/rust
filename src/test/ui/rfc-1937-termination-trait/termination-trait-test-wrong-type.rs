// compile-flags: --test

use std::num::ParseFloatError;

#[test]
fn can_parse_zero_as_f32() -> Result<f32, ParseFloatError> {
    //~^ ERROR the trait bound `f32: Termination` is not satisfied
    //~| ERROR the trait bound `f32: Termination` is not satisfied
    "0".parse()
}
