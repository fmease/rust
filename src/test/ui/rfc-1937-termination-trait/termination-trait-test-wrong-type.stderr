error[E0277]: the trait bound `f32: Termination` is not satisfied
  --> $DIR/termination-trait-test-wrong-type.rs:6:31
   |
LL | #[test]
   | ------- in this procedural macro expansion
LL | fn can_parse_zero_as_f32() -> Result<f32, ParseFloatError> {
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Termination` is not implemented for `f32`
   |
   = note: required for `Result<f32, ParseFloatError>` to implement `Termination`
note: required by a bound in `AssertIsTermination`
  --> $SRC_DIR/test/src/lib.rs:LL:COL
   |
LL | pub struct AssertIsTermination<T: Termination>(std::marker::PhantomData<T>);
   |                                   ^^^^^^^^^^^ required by this bound in `AssertIsTermination`
   = note: this error originates in the attribute macro `test` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `f32: Termination` is not satisfied
  --> $DIR/termination-trait-test-wrong-type.rs:6:1
   |
LL |   #[test]
   |   ------- in this procedural macro expansion
LL | / fn can_parse_zero_as_f32() -> Result<f32, ParseFloatError> {
LL | |
LL | |
LL | |     "0".parse()
LL | | }
   | |_^ the trait `Termination` is not implemented for `f32`
   |
   = note: required for `Result<f32, ParseFloatError>` to implement `Termination`
note: required by a bound in `assert_test_result`
  --> $SRC_DIR/test/src/lib.rs:LL:COL
   |
LL | pub fn assert_test_result<T: Termination>(result: T) -> Result<(), String> {
   |                              ^^^^^^^^^^^ required by this bound in `assert_test_result`
   = note: this error originates in the attribute macro `test` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
